src/share/poudriere/ports.sh

```diff
@@ -188,10 +188,14 @@ check_portsnap_interactive() {
 
 if [ ${CREATE} -eq 1 ]; then
 	# test if it already exists
-	porttree_exists ${PTNAME} && err 2 "The ports tree, ${PTNAME}, already exists"
-	maybe_run_queued "${saved_argv}"
 	: ${PTMNT="${BASEFS:=/usr/local${ZROOTFS}}/ports/${PTNAME}"}
 	: ${PTFS="${ZPOOL}${ZROOTFS}/ports/${PTNAME}"}
+	porttree_exists ${PTNAME}
+	if [ $? -eq 0 ]; then
+		run_hook ports create_done ${PTMNT} fail
```

Comment:
For hook names the pattern is more "hookname" "event", so these should just be "create fail", "create success", "update start", "update stop".

The `run_hook` command does export  some variables, but not _PTMNT_, so it is normal to pass this as the next argument.

https://github.com/freebsd/poudriere/wiki/hooks

See the others for examples, such as the "jail start" one.